### PR TITLE
fix: update tds details in tax withholding category

### PR DESCRIPTION
**Pull request by priyanshshah2442**
- Added Custom Fields in `Tax Withholding Category` Doctype:
  - `Section`
  - `Entity Type`
- Updated `tds_details.json` file
- Updating tds_section and entity_type based on old naming.
- Adding rules for financial year 24-25
- Validating only one withholding category exists for tds_section and entity_type.

### CODE REVIEW

File: india_compliance/income_tax_india/constants/custom_fields.py [hunk 1]
```diff
@@ -25,4 +25,26 @@
             ),
         }
     ],
+    "Tax Withholding Category": [
+        {
+            "label": "Section",
+            "fieldname": "section",
```
Comment (ljain112) [inline india_compliance/income_tax_india/constants/custom_fields.py]: fieldname can conflict with frappe,more precise

File: india_compliance/income_tax_india/constants/custom_fields.py [hunk 2]
```diff
@@ -25,4 +25,26 @@
...
+            "insert_after": "round_off_tax_amount",
+            "fieldtype": "Select",
```
Comment (ljain112) [inline india_compliance/income_tax_india/constants/custom_fields.py]: as there are multiple fields can we have autocomplete field ?

File: india_compliance/income_tax_india/data/tds_details.json [hunk 3]
```diff
@@ -1,48 +1,688 @@
 {
```
Comment (ljain112) [inline india_compliance/income_tax_india/data/tds_details.json]: more sections that can be added like 194q

File: india_compliance/income_tax_india/data/tds_details.json [hunk 4]
```diff
@@ -1,48 +1,688 @@
...
-        ["TDS - 194J - Director Fees - Individual", 10, 0, 0],
-        ["TDS - 194J - Director Fees - No PAN / Invalid PAN", 20, 0, 0]
-    ],
-    "Dividends": [
-        ["TDS - 194 - Dividends - Company", 10, 2500, 0],
-        ["TDS - 194 - Dividends - Individual", 10, 2500, 0],
-        ["TDS - 194 - Dividends - No PAN / Invalid PAN", 20, 2500, 0]
-    ]
-}
+    "2023": {
```
Comment (ljain112) [inline india_compliance/income_tax_india/data/tds_details.json]: we have to mention whole date so in future if any laws changes mid-year we can in corporate

File: india_compliance/income_tax_india/data/tds_details.json [hunk 5]
```diff
@@ -1,48 +1,1303 @@
...
-        ["TDS - 194J - Director Fees - Individual", 10, 0, 0],
-        ["TDS - 194J - Director Fees - No PAN / Invalid PAN", 20, 0, 0]
-    ],
-    "Dividends": [
-        ["TDS - 194 - Dividends - Company", 10, 2500, 0],
-        ["TDS - 194 - Dividends - Individual", 10, 2500, 0],
-        ["TDS - 194 - Dividends - No PAN / Invalid PAN", 20, 2500, 0]
-    ]
-}
+[
```
Comment (vorasmit) [inline india_compliance/income_tax_india/data/tds_details.json:1]: TDS on purchase seems to be missing

File: india_compliance/patches.txt [hunk 6]
```diff
@@ -40,3 +40,6 @@ india_compliance.patches.post_install.update_reconciliation_status
 india_compliance.patches.post_install.update_vehicle_no_field_in_purchase_receipt
 india_compliance.patches.post_install.update_gst_treatment_for_taxable_nil_transaction_item
 india_compliance.patches.post_install.update_default_gstr3b_status
+execute:from india_compliance.income_tax_india.setup import create_custom_fields; create_custom_fields()
+india_compliance.patches.v14.update_fields_in_tax_withholding_category
+india_compliance.patches.v14.update_tax_withholding_category
```
Comment (vorasmit) [inline india_compliance/patches.txt]: Post Install Patch could make sense

**Pull request merged**